ablySpec/RealtimeClientConnection.swift

```diff
@@ -1307,6 +1307,33 @@ class RealtimeClientConnection: QuickSpec {
 
             }
 
+            // RTN15
+            context("connection failures once CONNECTED") {
+
+                // RTN15g
+                it("when the connection resume has failed, all channels should be detached with an error reason") {
+                    let options = AblyTests.commonAppSetup()
+                    options.disconnectedRetryTimeout = 1.0
+
+                    let client = ARTRealtime(options: options)
+                    defer { client.close() }
+                    let channel = client.channels.get("test")
+
+                    waitUntil(timeout: testTimeout) { done in
+                        channel.attach() { errorInfo in
+                            expect(errorInfo).to(beNil())
+                            done()
+                        }
+                    }
+
+                    client.simulateLostConnection()
```

Comment:
As mentioned previously, can we change the name of this method because it's misleading.  Lost connection is not the same as lost connection state.  Please update to something like `simulateLostConnectionAndState`